core/src/layers/prometheus.rs

```diff
@@ -98,48 +96,84 @@ use crate::*;
 ///     Ok(())
 /// }
 /// ```
-#[derive(Default, Debug, Clone)]
+#[derive(Clone, Debug)]
 pub struct PrometheusLayer {
-    registry: Registry,
-    requests_duration_seconds_buckets: Vec<f64>,
-    bytes_total_buckets: Vec<f64>,
+    metrics: PrometheusMetricDefinitions,
+    /// The path level we will keep in the path label.
     path_label_level: usize,
 }
 
 impl PrometheusLayer {
-    /// create PrometheusLayer by incoming registry.
-    pub fn with_registry(registry: Registry) -> Self {
+    /// Create a [`PrometheusLayer`] while registering itself to this registry.
+    pub fn new(registry: Registry) -> Self {
+        Self::builder().build(registry)
+    }
+
+    /// Create a [`PrometheusLayerBuilder`].
+    pub fn builder() -> PrometheusLayerBuilder {
+        PrometheusLayerBuilder::default()
+    }
+}
+
+/// [`PrometheusLayerBuilder`] is a config builder to build a [`PrometheusLayer`].
+pub struct PrometheusLayerBuilder {
```

Comment:
I want to provide similar APIs for prometheus layer and prometheus-client layer,
however, the design of the `prometheus` library and the `prometheus-client` library is quite different, so I need to provide this builder pattern to construct a similar API.